src/file-selector.ts

```diff
@@ -108,34 +108,51 @@ function flatten<T>(items: any[]): T[] {
 async function fromDataTransferItem(
   item: DataTransferItem,
   entry?: FileSystemEntry | null,
-) {
-  // Check if we're in a secure context; due to a bug in Chrome (as far as we know) the browser crashes when calling this API (yet to be confirmed as a consistent behavior).
-  //
+): Promise<FileWithPath> {
+  const fileWithPath = await fromFileSystemHandle(item, entry);
```

Comment:
Coming from Go, these explicit variable names feel so long 😆 When a simple `f` could do.

Reply:
I prefer it explicit to be honest, more often than not when a block gets more complicated this starts feeling like you don't know what you are operating on any longer. E.g. below we also have a variable `file`, so `f` gets a bit muddled. That said, I could solve this much nicer if we had basic pattern matching in JS.